runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/PackageUtil.java

```diff
@@ -119,6 +119,8 @@ static PackageAttributes createPackageAttributes(File source,
       return new PackageAttributes(size, hash, directory, target, source.getPath());
     } catch (IOException e) {
       throw new RuntimeException("Package setup failure for " + source, e);
+    } finally {
+      hashStream.close();
```

Comment:
maybe close the `countingOutputStream` at L109?

* it's the outermost wrapper
* that will ensure the hasher has received all the bytes. (even though I happen to know that the counting stream does not buffer).